src/server/zset_family.cc

```diff
@@ -2279,6 +2279,51 @@ void ZSetFamily::ZRem(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void ZSetFamily::ZRandMember(CmdArgList args, ConnectionContext* cntx) {
+  string_view key = ArgS(args, 0);
+
+  ZRangeSpec range_spec;
+  range_spec.interval = IndexInterval(0, -1);
+  int count = 1;
+
+  switch (args.size()) {
+    case 3: {
+      ToUpper(&args[2]);
+      string_view arg = ArgS(args, 2);
+      range_spec.params.with_scores = arg == "WITHSCORES";
+      if (!range_spec.params.with_scores) {
+        return cntx->reply_builder()->SendError(absl::StrCat("unsupported option ", arg));
+      }
+      [[fallthrough]];
+    }
+    case 2:
+      if (string_view arg = ArgS(args, 1); !absl::SimpleAtoi(arg, &count)) {
+        cntx->reply_builder()->SendError(kInvalidIntErr);
+        return;
+      }
+      break;
+    case 1:
+      break;
+    default:
+      return cntx->reply_builder()->SendError(absl::StrCat("unsupported option ", ArgS(args, 3)));
+  }
+  bool sign = count < 0;
+  range_spec.params.limit = std::abs(count);
+
+  const auto cb = [&](Transaction* t, EngineShard* shard) {
+    return OpRange(range_spec, t->GetOpArgs(shard), key);
+  };
+
+  OpResult<ScoredArray> result = cntx->transaction->ScheduleSingleHopT(cb);
+  if (sign && result && !result->empty()) {
+    for (int i = range_spec.params.limit - result->size(); i > 0; --i) {
+      // we can return duplicate elements, so first is OK
+      result->push_back(result->front());
```

Comment:
it would take only one line to make it random for count > size by just choosing a random index from the original preix

Reply:
we don't use random for this function, so to my mind, it's better to be consistent and provide no random at all)